src/tcutility/cli_scripts/concatenate_irc.py

```diff
@@ -0,0 +1,185 @@
+
+    # Flatten the matrix and find the index of the minimum value
+    lowest_index = np.argmin(rmsd_matrix.flatten())
+
+    if lowest_index == 0:
+        _ = mols2.pop(0)
+        _ = energies2.pop(0)
+        return mols1[::-1] + mols2, energies1[::-1] + energies2
+    # elif lowest_index == 1:
+    #     _ = mols2.pop(-1)
+    #     _ = energies2.pop(-1)
+    #     return mols1 + mols2[::-1], energies1 + energies2[::-1]
+    # elif lowest_index == 2:
+    #     _ = mols2.pop(0)
+    #     _ = energies2.pop(0)
+    #     return mols2 + mols1[::-1], energies2 + energies1[::-1]
+    # elif lowest_index == 3:
+    #     _ = mols2.pop(-1)
+    #     _ = energies2.pop(-1)
+    #     return mols2[::-1] + mols1[::-1], energies2[::-1] + energies1[::-1]
+    else:  # This should never happen
+        raise ValueError("The RMSD values are not as expected.")
+
+
+def concatenate_irc_trajectories(job_dirs: list[str] | list[pl.Path], reverse: bool = False) -> tuple[list[Molecule], list[float]]:
+    """
+    Main function: concatenate trajectories from irc calculations, often being forward and backward, through the RMSD values.
+
+    Arguments:
+        job_dirs (list[str] | list[pl.Path]): List of directories containing the ams.rkf files.
+        reverse (bool): Reverse the trajectory. Default is False.
+
+    Returns:
+        tuple[list[Molecule], list[float]]: A tuple containing a list of Molecule objects and a list of energies.
+
+    Raises:
+        Exception: If an exception is raised in the try block, it is caught and printed.
+
+    """
+    job_dirs = [pl.Path(file) for file in job_dirs]
+    traj_geometries: list[list[Molecule]] = [[] for _ in job_dirs]
+    traj_energies: list[list[float]] = [[] for _ in job_dirs]
+    current_working_directory = pl.Path.cwd()
+
+    init(path=current_working_directory)
+    config.erase_workdir = True
+    config.log.stdout = 1
+
+    # This try / except / finally statement is to always remove the plams directory since we do not use it.
+    # Could be considered as an alternative to a contextmanager.
+    try:
+        for i, job_dir in enumerate(job_dirs):
+            print(f"Processing {job_dir}")
+            res = _create_ams_results(job_dir)
+            traj_geometries[i] = _get_converged_molecules(res)
+            traj_energies[i] = _get_converged_energies(res)
+    except Exception as e:
+        print("Got exception in concatenate_irc_trajectories:", e)
+    finally:
+        finish()
+
+    print("Concatenating trajectories...")
+    concatenated_mols, concatenated_energies = _concatenate_molecules_by_rmsd(traj_geometries[0], traj_geometries[1], traj_energies[0], traj_energies[1])
+
+    if reverse:
+        print("Reversing the trajectory...")
+        concatenated_mols = concatenated_mols[::-1]
+        concatenated_energies = concatenated_energies[::-1]
+    return concatenated_mols, concatenated_energies
+
+
+def _xyz_format(mol: Molecule) -> str:
```

Comment:
These molecule writing functions would be nice to have in the tcutility.molecule module